pkg/virt-api/rest/subresource.go

```diff
@@ -269,6 +269,46 @@ func getRunningJson(vm *v1.VirtualMachine, running bool) string {
 	}
 }
 
+func (app *SubresourceAPIApp) patchVmStart(vm *v1.VirtualMachine, namespace string, startPaused bool) error {
+	patchType := types.MergePatchType
+
+	if startPaused {
+		patchString := "{\"spec\": {\"template\": {\"spec\": {\"startStrategy\":  \"Paused\"}}}}"
+		log.Log.Object(vm).V(4).Infof("Patching VM: %s", patchString)
+		_, err := app.virtCli.VirtualMachine(namespace).Patch(vm.GetName(), patchType, []byte(patchString))
```

Comment:
I feel uneasy about permanently setting the startStrategy to paused on the VM when someone calls `virtctl start --paused my-vm`

My expectation if I called `virtctl start --paused my-vm`  is that the paused part would be a one time deal related to this one start command. If we mutate the startStrategy to be paused as well, then someone might accidently be committing to running with startStrategy paused forever.

So

`virtctl start --paused my-vm` - seems like i'd want to only run paused this one time.

Reply:
I agree. This is why paused startStrategy is removed when `virtctl stop my-vm` command is called. So the next start will not be affected by the previous one.

Reply:
> This is why paused startStrategy is removed when `virtctl stop my-vm` command is called

This assumes that `virtctl stop my-vm` is called before the next start, which isn't necessarily the case - the VMI might be deleted or the guest itself might shutdown voluntarily or just crash. In all of these cases, it would be automatically restarted in paused state, which might be surprising for the user.

I'd suggest to handle the `RunStrategyHalted` case similarly to the others - insert a `StateChangeRequest` with a `StartRequest` action, and let the `VMController` pick up the paused flag from there. You'll need to ensure that such a `StateChangeRequest` gets cleared by the controller (I think it already does, but need to ensure since this will have a slightly different code path within the controller).

Reply:
For this we will need to patch spec and status. Not sure it's feasible to do in 1 patch.

Reply:
> For this we will need to patch spec and status. Not sure it's feasible to do in 1 patch.

this would just be a status change. StartRequest+paused option.

Reply:
> this would just be a status change. StartRequest+paused option.

I believe @andreyod is referring to the case where `running: false` and `virtctl start --paused` is called. The existing behavior in that case is to set `running: true`, so if we want to maintain that , we'll need to update both spec and status and have a strategy to handle partial failures.

Reply:
> I believe @andreyod is referring to the case where `running: false` and `virtctl start --paused` is called.

Correct

Reply:
Changed the implementation to use status `StartRequest` for all cases.